extension/src/setup/studio.ts

```diff
@@ -142,17 +143,31 @@ export class Studio extends Disposable {
     verificationUrlWithParams.searchParams.append('code', userCode)
 
     await openUrl(verificationUrlWithParams.toString())
-    this.studioAccessTokenUriHandler = waitForUriResponse(
+    this.accessTokenUriHandler = waitForUriResponse(
       '/studio-complete-auth',
       () => {
         void this.requestStudioToken(deviceCode, tokenUri)
       }
     )
+    this.cancelUriHandlerAfterTimeout()
   }
 
   private resetAccessTokenUriHandler() {
-    this.studioAccessTokenUriHandler?.dispose()
-    this.studioAccessTokenUriHandler = undefined
+    if (this.accessTokenUriHandlerTimeout) {
+      clearTimeout(this.accessTokenUriHandlerTimeout)
+    }
+    this.accessTokenUriHandlerTimeout = undefined
+
+    this.accessTokenUriHandler?.dispose()
+    this.accessTokenUriHandler = undefined
+  }
+
+  private cancelUriHandlerAfterTimeout() {
+    const waitTime = 15 * 60000
+    this.accessTokenUriHandlerTimeout = setTimeout(
+      () => this.resetAccessTokenUriHandler(),
```

Comment:
I chose to fail silently, but we could also put up some kind of Toast message.

Reply:
Feels like 15 minutes is too long. I think 5 minutes would be a very reasonable amount of time to wait.

Reply:
Undecided on the toast message. I would lean towards showing it but I am conflicted.

Reply:
I'll set the time for five minutes. As for the toast message, I don't think it's needed for now. Most if not all users should just be authenticating immediately.